driver-core/src/main/java/com/datastax/driver/core/querybuilder/Clause.java

```diff
@@ -227,25 +229,32 @@ void appendTo(StringBuilder sb, List<Object> variables, CodecRegistry codecRegis
 
     static class CompoundInClause extends Clause {
         private final List<String> names;
-        private final List<?> valueLists;
+        private final List<Object> valueLists;
 
-        public CompoundInClause(List<String> names, List<?> valueLists) {
+        public CompoundInClause(Iterable<String> names, Iterable<?> valueLists) {
             if (valueLists == null)
                 throw new IllegalArgumentException("Missing values for IN clause");
-            if (valueLists.size() > 65535)
-                throw new IllegalArgumentException("Too many values for IN clause, the maximum allowed is 65535");
+            if (names == null)
+                throw new IllegalArgumentException("Missing names for IN clause");
+
+            this.names = Lists.newArrayList(names);
+            this.valueLists = Lists.newArrayListWithExpectedSize(this.names.size());
```

Comment:
Nit: should be `valueLists.size()`. The size of the list of tuples is not related to the number of values, for example `(a,b) IN ((1,2),(3,4),(5,6))` => 2 values, 3 tuples.
I'm taking care of this while merging.